applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -82,15 +82,9 @@ void BaseSolidElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
 
 void BaseSolidElement::InitializeSolutionStep( const ProcessInfo& rCurrentProcessInfo )
 {
-    // We initialize the material reponse if required
-    bool required = false;
-    for ( IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number ) {
-        if (mConstitutiveLawVector[point_number]->RequiresInitializeMaterialResponse()) {
-            required = true;
-            break;
-        }
-    }
+    const bool required = mConstitutiveLawVector[0]->RequiresInitializeMaterialResponse();
```

Comment:
you can have different CLs within the element...i think that the previosu code was better here

Reply:
ok! i did not think that this could happen